### PR TITLE
Added hiddenfield

### PR DESCRIPTION
**Pull request by rosscdh**
Thanks for contributing to Wagtail! 🎉

Before submitting, please review the contributor guidelines <http://docs.wagtail.io/en/latest/contributing/index.html> and check the following:

* Do the tests still pass? (http://docs.wagtail.io/en/latest/contributing/developing.html#testing): yes

* Does the code comply with the style guide? (Run `make lint` from the Wagtail root): no but these seem like unrelated pre-existing lint issues

* For Python changes: Have you added tests to cover the new/fixed behaviour?: yes

```
python runtests.py wagtail.wagtailforms                                         (feature/form-hidden-field-for-js-population)
Creating test database for alias 'default'...
..................................................................
----------------------------------------------------------------------
Ran 66 tests in 7.016s
```

* For new features: Has the documentation been updated accordingly?: dont appear to be any for form fields?

### CODE REVIEW

File: wagtail/wagtailforms/forms.py [hunk 1]
```diff
@@ -81,6 +84,7 @@ def create_checkbox_field(self, field, options):
         'radio': create_radio_field,
         'checkboxes': create_checkboxes_field,
         'checkbox': create_checkbox_field,
+        'hiddenfield': create_hidden_field,
```
Comment (zerolab) [inline wagtail/wagtailforms/forms.py]: `hidden` as the field type is more in line with the other declarations.

Comment (rosscdh) [inline wagtail/wagtailforms/forms.py]: +1

**Comment by rosscdh** [comment]
ready for review?

**Comment by BertrandBordage** [comment]
I don’t get it: what is the point of adding hidden field to the form builder, since users won’t be able to fill it?

**Comment by rosscdh** [comment]
It's a fair question.
Use case:

As a form creator
I sometimes need to pass in hidden data to a form to allow for sub
categorisation or handling of items that may relate to that hidden field
value.

For example we have 5000 games. All basically the same except for a hidden
field value that will send data to a 3rd party api. Or maybe show a
different logo or something similar.

This hidden field would go a long way to making that kind of flexibility
where u don't want 5000 basically the same game models.

Hope I've explained that in a non superfluous way.

**Comment by gasman** [comment]
Hi @rosscdh - sorry for the delay in reviewing.

It looks like the new field type still needs adding to FORM_FIELD_CHOICES in wagtailforms/models.py, so that it shows up in the list for users to choose from.

**Comment by rosscdh** [comment]
Thanks Ill try get to it this weekend.. was a little slow but thanks for
doing it

**Comment by rosscdh** [comment]
Added as requested. Tests failing due to unrelated system problems (trying to install oracle java 8)

**Comment by gasman** [comment]
Hi @rosscdh - the tests are failing because the change to the choices list has changed the models in the test app, and so they now need a migration. You should be able to generate it by running `django-admin.py makemigrations --settings=wagtail.tests.settings` from the Wagtail root.

**Comment by rosscdh** [comment]
thanks will do.. Strange the error i saw on travis said was to do with
oracle.

**Comment by rosscdh** [comment]
Strange the contributing guideline instructions followed and I am now
getting:



(test_wagta) ±|feature/form-hidden-field-for-js-population ✔| wagtail
% django-admin makemigrations --settings=wagtail.tests.settings
                                 rosscdh@s
Traceback (most recent call last):
  File "/usr/local/bin/django-admin", line 11, in <module>
    sys.exit(execute_from_command_line())
  File
"/Library/Python/2.7/site-packages/django/core/management/__init__.py",
line 364, in execute_from_command_line
    utility.execute()
  File
"/Library/Python/2.7/site-packages/django/core/management/__init__.py",
line 338, in execute
    django.setup()
  File "/Library/Python/2.7/site-packages/django/__init__.py", line 27, in
setup
    apps.populate(settings.INSTALLED_APPS)
  File "/Library/Python/2.7/site-packages/django/apps/registry.py", line
108, in populate
    app_config.import_models()
  File "/Library/Python/2.7/site-packages/django/apps/config.py", line 202,
in import_models
    self.models_module = import_module(models_module_name)
  File
"/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py",
line 37, in import_module
    __import__(name)
  File
"/usr/local/lib/python2.7/site-packages/wagtail/tests/testapp/models.py",
line 20, in <module>
    from wagtail.wagtailcore.models import Page, Orderable, PageManager
  File
"/usr/local/lib/python2.7/site-packages/wagtail/wagtailcore/models.py",
line 269, in <module>
    class Page(six.with_metaclass(PageBase, MP_Node, ClusterableModel,
index.Indexed)):
  File "/Library/Python/2.7/site-packages/django/utils/six.py", line 808,
in __new__
    return meta(name, bases, d)
  File
"/usr/local/lib/python2.7/site-packages/wagtail/wagtailcore/models.py",
line 235, in __init__
    if cls._deferred:
AttributeError: type object 'Page' has no attribute '_deferred'


% pip freeze
                                rosscdh@s
beautifulsoup4==4.6.0
boto3==1.4.7
botocore==1.7.25
certifi==2017.7.27.1
chardet==3.0.4
configparser==3.5.0
coverage==4.4.1
Django==1.11.6
django-modelcluster==3.1
django-taggit==0.22.1
django-treebeard==4.1.2
djangorestframework==3.7.0
docutils==0.14
elasticsearch==2.4.1
enum34==1.1.6
flake8==3.4.1
flake8-blind-except==0.1.1
flake8-print==2.0.2
freezegun==0.3.9
funcsigs==1.0.2
futures==3.1.1
html5lib==0.999999999
idna==2.6
isort==4.2.5
Jinja2==2.9.6
jmespath==0.9.3
MarkupSafe==1.0
mccabe==0.6.1
mock==2.0.0
olefile==0.44
pbr==3.1.1
Pillow==4.3.0
pycodestyle==2.3.1
pyflakes==1.5.0
python-dateutil==2.6.1
pytz==2017.2
requests==2.18.4
s3transfer==0.1.11
six==1.11.0
Unidecode==0.4.21
urllib3==1.22
webencodings==0.5.1
Willow==0.4

**Comment by gasman** [comment]
@rosscdh Looks like it's picking up an old installation of Wagtail there - that's the error you'd generally see when running Django >=1.10 against a release of Wagtail that didn't support it yet. The fact that the stack trace shows references to Wagtail in `site-packages` rather than your working copy of Wagtail is suspicious too...

Running `pip uninstall wagtail` then `pip install -e .` from the root of your working copy might fix things.

**Comment by gasman** [comment]
Migration now added, and merged in 32555f7a1c599c139e0f26c22907c9612af2e015 - thanks @rosscdh!

**Comment by rosscdh** [comment]
My pleasure guys thanks for the awesome product.

**Pull request closed without merging**